activesupport/lib/active_support/encrypted_configuration.rb

```diff
@@ -43,6 +43,12 @@ def message
       end
     end
 
+    class InvalidKeyError < RuntimeError
+      def initialize(content_path, key)
+        super "Invalid key '#{key}' from configuration in '#{content_path}'."
```

Comment:
I like that we're raising a more informative error message, but I think this error can be clearer that the key is invalid _because_ it is an integer and therefore it can't be deserialized. `Invalid key` reads like "key does not exist" to me.

Reply:
@eileencodes Thanks for the feedback :bow:

I've updated the message to:

```
ActiveSupport::EncryptedConfiguration::InvalidKeyError: Key '42' is invalid, it must not be an Integer or Float, and respond to '#to_sym' from configuration in '/tmp/config-20240530-4153925-tfzzt/credentials.yml.enc'.
```

It felt long, but I realized there are other cases (including Float) where user can try to use a key in the YAML which is valid but doesn't respond to `#to_sym`. Another case I did not cover is boolean (e.g. `On: value`) would raise this error.

Should the list be exhaustive? Or something simple like:

```
Key '#{key}' (#{key.class}) does not respond to '#to_sym' from configuration in '/tmp/config/....yml.enc'.
```

Reply:
I think the second one is better. No need to list all the classes we do or don't accept, knowing it doesn't respond to to_sym is enough.